stratum/lib/security/credentials_manager.cc

```diff
@@ -63,95 +101,73 @@ CredentialsManager::GenerateExternalFacingClientCredentials() const {
   return client_credentials_;
 }
 
-::util::Status CredentialsManager::Initialize(bool secure_only) {
-  // Server credentials.
+::util::Status CredentialsManager::Initialize() {
+  RETURN_IF_ERROR(InitializeServerCredentials());
+  return InitializeClientCredentials();
+}
+
+::util::Status CredentialsManager::InitializeServerCredentials() {
   if (FLAGS_ca_cert_file.empty() && FLAGS_server_key_file.empty() &&
       FLAGS_server_cert_file.empty()) {
-    if (secure_only) {
-      LOG(WARNING) << "No certificate/key files provided, cannot initiate gRPC "
-                      "server credentials";
-      server_credentials_ = nullptr;
+    // No server credentials files
+    if (secure_only_) {
+      LOG(WARNING) << "No key/certificate files specified; "
+                      "cannot initialize server credentials.";
     } else {
-      LOG(WARNING) << "No certificate/key files provided, using insecure "
-                      "server credentials!";
+      LOG(WARNING) << "No key/certificate files specified; "
+                      "using insecure server credentials!";
       server_credentials_ = ::grpc::InsecureServerCredentials();
     }
+  } else if (secure_only_ && !ServerFilesOk()) {
```

Comment:
Code looks ok, but should be tested to make sure that there is no fallbacks to insecure modes